File: test/language/templates/python/InstantiateTypeAsTemplateArgumentTest.py
import zserio

import Templates

class InstantiateTypeAsTemplateArgumentTest(Templates.TestCase):
    def testReadWrite(self):
        instantiateTypeAsTemplateArgument = self.api.InstantiateTypeAsTemplateArgument(
            self.api.Other_Str(self.api.Str("test"))
        )

        writer = zserio.BitStreamWriter()
        instantiateTypeAsTemplateArgument.write(writer)
        reader = zserio.BitStreamReader(writer.byte_array, writer.bitposition)
        readInstantiateTypeAsTemplateArgument = self.api.InstantiateTypeAsTemplateArgument()
        readInstantiateTypeAsTemplateArgument.read(reader)
        self.assertEqual(instantiateTypeAsTemplateArgument, readInstantiateTypeAsTemplateArgument)

File: test/language/templates/cpp/InstantiateTypeAsTemplateArgumentTest.cpp
#include "gtest/gtest.h"

#include "templates/instantiate_type_as_template_argument/InstantiateTypeAsTemplateArgument.h"

namespace templates
{
namespace instantiate_type_as_template_argument
{

TEST(InstantiateTypeAsTemplateArgumentTest, readWrite)
{
    InstantiateTypeAsTemplateArgument instantiateTypeAsTemplateArgument{
            Other_Str{Str{"test"}}
    };

    zserio::BitBuffer bitBuffer = zserio::BitBuffer(1024 * 8);
    zserio::BitStreamWriter writer(bitBuffer);
    instantiateTypeAsTemplateArgument.write(writer);

    zserio::BitStreamReader reader(writer.getWriteBuffer(), writer.getBitPosition(), zserio::BitsTag());
    InstantiateTypeAsTemplateArgument readInstantiateTypeAsTemplateArgument(reader);

    ASSERT_TRUE(instantiateTypeAsTemplateArgument == readInstantiateTypeAsTemplateArgument);
}

} // namespace instantiate_type_as_template_argument
} // namespace templates
